lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -149,7 +149,7 @@ public function listTableForeignKeys($table, $database = null)
                 $id = $value['id'];
 
                 $tableForeignKeys[$key] = array_merge($tableForeignKeys[$key], [
-                    'constraint_name' => isset($names[$id]) && $names[$id] !== '' ? $names[$id] : null,
+                    'constraint_name' => isset($names[$id]) && $names[$id] !== '' ? $names[$id] : "this_is_fun_$id",
```

Comment:
Could we detect numeric constraint names starting at 0 and make the name `NULL`? That would skip the constraint getting a name during `AbstractPlatform::getForeignKeyBaseDeclarationSQL`.

Reply:
It's kind of what we have right now since https://github.com/doctrine/dbal/pull/4246, and that results in a bug where 2 constraints become one. I described it here: https://github.com/doctrine/dbal/issues/4243#issuecomment-687362580